crates/wasmtime/src/runtime/module.rs

```diff
@@ -1027,6 +1027,27 @@ impl Module {
         })
     }
 
+    /// Get the locations of functions in this module's `.text` section, with the
+    /// function names.
+    ///
+    /// Each function's location is a (name, `.text` section offset, length) pair.
+    pub fn function_locations_with_names<'a>(
+        &'a self,
+    ) -> impl ExactSizeIterator<Item = (String, usize, usize)> + 'a {
+        let module = self.compiled_module();
+        self.function_locations()
+            .enumerate()
+            .map(|(idx, (offset, len))| {
+                let idx = DefinedFuncIndex::from_u32(u32::try_from(idx).unwrap());
+                let idx = module.module().func_index(idx);
+                if let Some(name) = module.func_name(idx) {
+                    (format!("function[{}]::{}", idx.as_u32(), name), offset, len)
+                } else {
+                    (format!("function[{}]", idx.as_u32()), offset, len)
+                }
```

Comment:
Instead of stringifying the name with the index here, and then forcing the consumer of this API in `crates/explore/src/lib.rs` to parse the name out from stringified name out, can we just give the caller the various components before stringification? That is, return an iterator that yields items of type

```rust
(
    u32,            // wasm function index
    Option<String>, // name, if available
    usize,          // text section offset
    usize,          // length of function in text section
)
```

I think we could also update the existing `function_locations` method, and perhaps rename it, rather than defining a whole new method, since that method exists to basically enable `wasmtime-explore`.

Reply:
Alright.